src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer.java

```diff
@@ -88,7 +88,7 @@ public void run(RunConfiguration config) throws AppEngineException {
     arguments.addAll(DevAppServerArgs.get("runtime", config.getRuntime()));
     arguments.addAll(DevAppServerArgs.get("allow_skipped_files", config.getAllowSkippedFiles()));
     arguments.addAll(DevAppServerArgs.get("api_port", config.getApiPort()));
-    arguments.addAll(DevAppServerArgs.get("automatic_restart", config.getAutomaticRestart()));
+    arguments.addAll(DevAppServerArgs.get("automatic_restart", config.getAutomaticRestart(), true));
```

Comment:
The way I see it, the real issue is fundamentally `--automatic_restart` being `true` by default in dev_appserver. My preferred fix would be to just default it to false on dev_appserver and move on, but that change would take longer. (Manifestly, I don't know why automatic restart defaults to true, or can think of any case where that behaviour is helpful)

That being said, I like that we're making up in the library for that issue in dev_appserver, and prefer we stick to default false.

Regarding your first point, checking the state of the other dev_appserver flags seems to be a popular request these days, so I'll go ahead and do that in the scope of this PR.